src/Module.php

```diff
@@ -4,38 +4,28 @@
 
 namespace Laminas\Hydrator;
 
-use Laminas\ModuleManager\ModuleManager;
+use Laminas\ServiceManager\ServiceManager;
 
-class Module
+/**
+ * @psalm-import-type ServiceManagerConfiguration from ServiceManager
+ */
+final class Module
 {
     /**
      * Return default laminas-hydrator configuration for laminas-mvc applications.
      *
-     * @return mixed[]
+     * @return array[]
+     * @psalm-return array{service_manager: ServiceManagerConfiguration}
      */
     public function getConfig(): array
     {
         $provider = new ConfigProvider();
 
+        /** @var ServiceManagerConfiguration $config */
```

Comment:
It would be better if you could amend `ConfigProvider` to return `ServiceManagerConfiguration`, then the type would not need to be forced 👍